src/tools/illink/src/ILLink.Tasks/ILLink.Tasks.csproj

```diff
@@ -78,4 +78,23 @@
     </ProjectReference>
   </ItemGroup>
 
+  <Target Name="GenerateBuildMetadata" BeforeTargets="BeforeBuild">
```

Comment:
Please add inputs and outputs to the target. Input would be MSBuildProjectFile and output the intermediate path.